src/__tests__/utils/runForRenderers.ts

```diff
@@ -0,0 +1,86 @@
+import {
+  ReactHooksRenderer,
+  ReactHooksServerRenderer,
+  RenderHookOptions,
+  RenderHookResult
+} from '../../types/react'
+
+type RendererResolvers = typeof rendererResolvers
+type Renderer = keyof RendererResolvers
+type InferredRenderer<TRenderers extends Renderer> = ReturnType<RendererResolvers[TRenderers]>
+
+declare global {
+  function runForRenderers<TRenderers extends Renderer>(
+    renderers: TRenderers[],
+    fn: (renderer: InferredRenderer<TRenderers>, rendererName: Renderer) => void
+  ): void
+
+  function runForLazyRenderers<TRenderer extends Renderer>(
+    renderers: TRenderer[],
+    fn: (getRenderer: () => InferredRenderer<TRenderer>, rendererName: Renderer) => void
+  ): void
+}
+
+function requireRenderer<TRendererType extends ReactHooksRenderer = ReactHooksRenderer>(
+  rendererName: Renderer
+) {
+  let requirePath = `../../${rendererName}`
+  if (rendererName.startsWith('default')) {
+    requirePath = requirePath.replace('/default', '')
+  }
+  /* eslint-disable @typescript-eslint/no-var-requires */
+  return require(requirePath) as TRendererType
+}
+
+function hydratedServerRenderer(pure: boolean): ReactHooksRenderer {
+  const { renderHook, ...otherImports } = requireRenderer<ReactHooksServerRenderer>(
+    pure ? 'server/pure' : 'server'
+  )
+
+  return {
+    renderHook<TProps, TResult>(
+      callback: (props: TProps) => TResult,
+      options?: RenderHookOptions<TProps>
+    ): RenderHookResult<TProps, TResult> {
+      const { hydrate, ...otherUtils } = renderHook<TProps, TResult>(callback, options)
+      hydrate()
+      return {
+        ...otherUtils
+      }
+    },
+    ...otherImports
+  }
+}
```

Comment:
maybe worth writing this as an inline comment so _others_ know?

Reply:
To clarify, a comment saying why this renderer only exists in the test suite or when you should use it in tests, or both?

I've also been wondering if there is any value in moving this renderer into the `src` and exporting it from the library as well. I'm not sure when it would be used over the `dom` renderer though?

Reply:
I've added something. Let me know if there's anything else that should be added.

Reply:
That looks good ⭐